src/main/java/seedu/budgetbuddy/transaction/saving/SavingsManager.java

```diff
@@ -0,0 +1,118 @@
+package seedu.budgetbuddy.transaction.saving;
+
+import seedu.budgetbuddy.transaction.expense.Expense;
+import seedu.budgetbuddy.transaction.expense.ExpenseManager;
+import seedu.budgetbuddy.transaction.income.Income;
+import seedu.budgetbuddy.transaction.income.IncomeManager;
+import seedu.budgetbuddy.util.LoggerSetup;
+
+import java.time.YearMonth;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.logging.Logger;
+
+/**
+ * Displays the savings of the user.
+ * The user can specify if they wish to see their total savings by month or in total.
+ */
+public class SavingsManager {
+    private static final Logger LOGGER = LoggerSetup.getLogger();
+
+    /**
+     * Displays the total savings of the user based on all their incomes and expenses.
+     * Total Savings = Total Income - Total Expenses
+     */
+    public static String displayTotalSavings(){
+        double savings = 0;
+        YearMonth firstIncome = YearMonth.now();
+        YearMonth firstExpense = YearMonth.now();
+        String result = "";
+        if (IncomeManager.getIncomes().size() > 0){
+            for (Income income: IncomeManager.getIncomes()){
+                savings += income.getAmount();
+                YearMonth incomeYearMonth = IncomeManager.getYearMonthFromDate(income.getDate());
+                if (incomeYearMonth.compareTo(firstIncome) == -1){
+                    firstIncome = incomeYearMonth;
+                }
+            }
+        }
```

Comment:
looks deeply nested. can you try to use some guard clause to eliminate the nesting if possible?